src/horizontalaligner.cpp

```diff
@@ -33,6 +33,97 @@
 
 namespace vrv {
 
+//----------------------------------------------------------------------------
+// Fraction
+//----------------------------------------------------------------------------
+
+Fraction::Fraction(int num, int denom) : m_numerator(num), m_denominator(denom)
+{
+    if (denom == 0) {
+        LogDebug("Denominator cannot be zero.");
+        denom = 1;
+    }
+    Reduce();
+}
+
+Fraction Fraction::operator+(const Fraction &other) const
+{
+    int num = m_numerator * other.m_denominator + other.m_numerator * m_denominator;
+    int denom = m_denominator * other.m_denominator;
+    return Fraction(num, denom);
+}
+
+Fraction Fraction::operator-(const Fraction &other) const
+{
+    int num = m_numerator * other.m_denominator - other.m_numerator * m_denominator;
+    int denom = m_denominator * other.m_denominator;
+    return Fraction(num, denom);
+}
+
+Fraction Fraction::operator*(const Fraction &other) const
+{
+    int num = m_numerator * other.m_numerator;
+    int denom = m_denominator * other.m_denominator;
+    return Fraction(num, denom);
+}
+
+Fraction Fraction::operator/(const Fraction &other) const
+{
+    if (other.m_numerator == 0) {
+        LogDebug("Cannot divide by zero.");
+        return *this;
+    }
+    int num = m_numerator * other.m_denominator;
+    int denom = m_denominator * other.m_numerator;
+    return Fraction(num, denom);
+}
+
+bool Fraction::operator==(const Fraction &other) const
+{
+    return (m_numerator == other.m_numerator) && (m_denominator == other.m_denominator);
+}
+
+bool Fraction::operator<(const Fraction &other) const
+{
+    return m_numerator * other.m_denominator < other.m_numerator * m_denominator;
+}
+
+bool Fraction::operator<=(const Fraction &other) const
+{
+    return m_numerator * other.m_denominator <= other.m_numerator * m_denominator;
+}
+
+bool Fraction::operator>(const Fraction &other) const
+{
+    return m_numerator * other.m_denominator > other.m_numerator * m_denominator;
+}
+
+bool Fraction::operator>=(const Fraction &other) const
+{
+    return m_numerator * other.m_denominator >= other.m_numerator * m_denominator;
+}
+
+double Fraction::ToDouble() const
+{
+    return static_cast<double>(m_numerator) / m_denominator;
+}
+
+std::string Fraction::ToString() const
+{
+    return StringFormat("%d/%d", m_numerator, m_denominator);
+}
+
+void Fraction::Reduce()
+{
+    if (m_denominator < 0) { // Keep the denominator positive
+        m_numerator = -m_numerator;
+        m_denominator = -m_denominator;
+    }
+    int gcdVal = std::gcd(abs(m_numerator), abs(m_denominator));
+    m_numerator /= gcdVal;
+    m_denominator /= gcdVal;
```

Comment:
Maybe add an if statement to only divide by gcdVal if it is not 1 (to save time of two unnecessary divisions in that case).